### PR TITLE
MAYA-110155: consistent namespace across the project - Part 1

### PR DESCRIPTION
**Pull request by HamedSabri-adsk**
We currently don't have a consistent namespace under "lib/mayaUsd" and in many places we are not respecting the namespaces at all making it hard to read and write code logic. This PR is the first of many to do this cleanup.

### CODE REVIEW

File: lib/mayaUsd/nodes/layerManager.cpp [hunk 1]
```diff
@@ -171,7 +171,7 @@ void convertAnonymousLayersRecursive(
             convertAnonymousLayersRecursive(subL, basename, stage);
 
             if (subL->IsAnonymous()) {
-                auto newLayer = UsdMayaSerialization::saveAnonymousLayer(subL, layer, basename);
+                auto newLayer = MayaUsd::utils::saveAnonymousLayer(subL, layer, basename);
```
[the remark this answers is not in the document]
Comment (hamedsabri) [reply lib/mayaUsd/nodes/layerManager.cpp:174]: Picking small unit names for namespace makes thing a bit complicated the way maya-usd project is structured. Let me explain what I mean by small unit names:

Under mayaUsd/utils, we have serialization, filesystem, util, etc...

These are tiny utility classes and using them for "namespace grouping" could be confusing. If you think about it, they are not really big units.

MayaUsd::serialization::
MayaUsd::filesystem::
MayaUsd::util::

We also want to minimize the number of namespace. Having a lot of namespaces could be confusing. 
We could also add more nested namespaces ( if really needed ). 

I think going with bigger unit to start a namespace after "MayaUsd::" makes more sense.

```
MayaUsd::base::
MayaUsd::commands::
MayaUsd::fileio::
MayaUsd::listeners::
MayaUsd::nodes::
MayaUsd::python::
MayaUsd::render::
MayaUsd::ufe::
MayaUsd::undo::
MayaUsd::utils::
```

[the remark this answers is not in the document]
Comment (ppt-adsk) [reply lib/mayaUsd/nodes/layerManager.cpp:174]: Works for me, and I agree with your point of too many fine-grained namespaces.  If others are fine with it, it's a go.

**Review by ppt-adsk — APPROVED** [review]

File: lib/mayaUsd/nodes/layerManager.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (seando-adsk) [reply lib/mayaUsd/nodes/layerManager.cpp:174]: I don't think we need fine-grained namespaces. Even just a single MayaUsd namespace would be okay with me.

[the remark this answers is not in the document]
Comment (HamedSabri-adsk) [reply lib/mayaUsd/nodes/layerManager.cpp:174]: > just a single MayaUsd namespace

No... that makes things very complicated specially with the way codes are structured in this particular project.

Let's say I have a tomato() under two different modules under mayaUsd and I want to use it all over the project.

e.g

```
lib/mayaUsd/A/fooUtil.h
void tomato();

lib/mayaUsd/B/fooUtil.h
void tomato();
```

Then I pepper the code with this:

`MayaUsd::tomato();`

Hmm, whose tomato is this again??

But if I go with the Module as a namespace ( A or B ), the code reads a lot better and can be discovered much easier.

```
MayaUsd::A::tomato();
MayaUsd::B::tomato();
```

Again, namespace is all about grouping/modularizing/organizing bunch of code.

In my past experience in a different team, we had a very restrict rule and clear namespaces. I can even remember it now without thinking hard.

```
Project
dataModel
    alg   ( namespace akDMA )    data model algorithm
    defn  ( namespace akDMD )    data model definition
    gui   ( namespace akDMG )    data model graphic


akDMA::foo();
akDMD::foo();
akDMG::foo();
```

As I said before, I am not too crazy about nested namespaces. To me 1 or 2 level deep is good enough depending how the codes are structured.

**Pull request merged**